include/samurai/load_balancing.hpp

```diff
@@ -0,0 +1,419 @@
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+#include "algorithm.hpp"
+#include "algorithm/utils.hpp"
+#include "mesh.hpp"
+#include "mr/mesh.hpp"
+#include "timers.hpp"
+
+#ifdef SAMURAI_WITH_MPI
+#include <boost/mpi.hpp>
+#endif
+
+#ifdef SAMURAI_WITH_MPI
+namespace samurai
+{
+    enum BalanceElement_t
+    {
+        CELL
+    };
+
+    class Weight
+    {
+      public:
+
+        template <class Field>
+        static auto from_field(const Field& f)
+        {
+            auto weight = samurai::make_scalar_field<double>("weight", f.mesh());
+            weight.fill(0.);
+            samurai::for_each_cell(f.mesh(),
+                                   [&](auto cell)
+                                   {
+                                       weight[cell] = f[cell];
+                                   });
+            return weight;
+        }
+
+        template <class Mesh>
+        static auto uniform(const Mesh& mesh)
+        {
+            auto weight = samurai::make_scalar_field<double>("weight", mesh);
+            weight.fill(1.);
+
+            return weight;
+        }
+
+        template <BalanceElement_t elem, class Mesh_t, class Field_t>
+        static double compute_load(const Mesh_t& mesh, const Field_t& weight)
+        {
+            using mesh_id_t             = typename Mesh_t::mesh_id_t;
+            const auto& current_mesh    = mesh[mesh_id_t::cells];
+            double current_process_load = 0.;
+            // cell-based load with weight.
+            samurai::for_each_cell(current_mesh,
+                                   [&](const auto& cell)
+                                   {
+                                       current_process_load += weight[cell];
+                                   });
+            return current_process_load;
+        }
+    };
+
+    template <class Flavor>
+    class LoadBalancer
```

Comment:
It seems that all the element of this class are public. Maybe we can change it as `struct` and remove the `public` keyword.